.github/workflows/CI.yml

```
@@ -272,7 +272,7 @@ jobs:
     needs:
       - test-unix
       - test-cygwin
-    if: ${{ always() && github.event_name == 'push' && github.repository == 'gap-system/gap' }}
+    if: ${{ always() && github.event_name != 'pull_request' && github.repository == 'gap-system/gap' }}
```

Actually can't we even write this?
```suggestion
    if: ${{ github.event_name != 'pull_request' && github.repository == 'gap-system/gap' }}
```

---

No, it is necessary, the `needs:` context implies by default that the `slack-notification-on-failure` job (which I will rename) would only run if `test-unix` and `test-cygwin` are _successful_

https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-requiring-dependent-jobs-to-be-successful

---

The documentation also says that "always()" simply always returns `true`. There is nothing magical about it, as gar as I could tell

---

Ah right, I see, thanks for the further explanation. I'll give it a go.

---

It seems that `always()` _is_ magical.

In a branch on my fork I broke the tests and removed the `always()`. Without the `always()`, the Slack job was skipped: https://github.com/wilfwilson/gap/actions/runs/795815484 but when I changed nothing except to add back the `always()`, the Slack job was run https://github.com/wilfwilson/gap/actions/runs/795831635, as it should be.

---

That must mean that, in the absence of an `always()`, the GitHub Actions software implicitly adds the condition `!failure() &&` to the `if:` line.

---

OK, sorry for wasting your time with this then -- and I guess the GitHub docs need to be improved :/